crates/aranya-fast-channels/src/shm/read.rs

```diff
@@ -138,112 +72,54 @@ where
         //
         // NB: we load the generation before traversing the list
         // to avoid ownership conflicts with `chan`.
-        let generation = list.generation.load(Ordering::Relaxed);
+        let _generation = list.generation.load(Ordering::Relaxed);
 
-        let (chan, idx) = match list.find_mut(id, hint, Op::Seal)? {
+        // TODO: Don't eagerly search list. Compare to generation
+        let (chan, _idx) = match list.find_mut(id, None, Op::Seal)? {
             None => return Err(crate::Error::NotFound(id)),
             Some((chan, idx)) => (chan, idx),
         };
 
-        let mut key = SealKey::from_raw(&chan.seal_key, chan.seq())?;
+        let label_id = *ctx.label_id();
```

Comment:
Seems clearer to have `label_id()` return a copy of `LabelID` instead of a reference to it that is dereferenced later. I noticed a push towards using IDs by value in the repos recently so it seems like that would apply here as well. cc: @jdygert-spok 
```suggestion
        let label_id = ctx.label_id();
```
```
        /// Returns the label ID.
        pub fn label_id(&self) -> LabelId {
            self.label
        }
```